src/uct/ib/rc/accel/gga_mlx5.c

```diff
@@ -24,11 +27,41 @@ typedef struct {
     uct_rc_mlx5_iface_common_t  super;
 } uct_gga_mlx5_iface_t;
 
-typedef struct uct_gga_mlx5_iface_config {
+typedef struct {
     uct_rc_iface_config_t             super;
     uct_rc_mlx5_iface_common_config_t rc_mlx5_common;
 } uct_gga_mlx5_iface_config_t;
 
+typedef struct {
+    uint8_t       buf[UCT_GGA_MLX5_OPAQUE_BUF_LEN];
+    struct ibv_mr *mr;
+} UCS_V_ALIGNED(UCS_SYS_CACHE_LINE_SIZE) uct_gga_mlx5_dma_opaque_buf_t;
+
+typedef struct {
+    uct_rc_mlx5_iface_common_qp_cleanup_ctx_t super;
+    uct_gga_mlx5_dma_opaque_buf_t             dma_opaque;
+} uct_gga_mlx5_iface_qp_cleanup_ctx_t;
+
+typedef struct {
+    uct_rc_mlx5_base_ep_t         super;
+    uct_gga_mlx5_dma_opaque_buf_t dma_opaque;
+} uct_gga_mlx5_ep_t;
+
+enum {
+    UCT_GGA_MLX5_EP_ADDRESS_FLAG_FLUSH_RKEY = UCS_BIT(0)
+};
+
+typedef struct {
+    uint8_t         flags;
+    uct_ib_uint24_t qp_num;
+    uint32_t        flush_rkey;
+} UCS_S_PACKED uct_gga_mlx5_ep_address_t;
+
+typedef struct {
```

Comment:
it seems GGA transport returns same device name as all other IB transports. so from UCT API perspective, it reuses the same UCT "device". therefore, we cannot return a different IB address here - it may confuse ucp_worker_address_pack to group rc,ud,dc,.. and gga under the same device address

can we move sys_image_guid to iface address instead?

Reply:
done